File: Data Structures & Algorithms/validate-parentheses/submission-5.py
class Solution:
    def isValid(self, s: str) -> bool:
        paren = {')': '(', '}': '{', ']': '['}
        stack = list()

        for ch in s:
            if ch in paren:
                if not stack or paren[ch] != stack.pop():
                    return False
            else:
                stack.append(ch)

        return not stack

File: Data Structures & Algorithms/daily-temperatures/submission-5.swift
class Solution{
    func dailyTemperatures(_ temperatures: [Int]) -> [Int] {
        var res: [Int] = Array(repeating: 0, count: temperatures.count)
        var stack: [Int] = []

        for i in 0...(temperatures.count-1){
            while !stack.isEmpty && temperatures[i] > temperatures[stack.last!]{
                res[stack.last!] = i - stack.removeLast()
            }
            stack.append(i)
        }
        return res
    }
}